### ISSUE TITLE
Invalid Projection EPSG:102008

### ISSUE THREAD

**Issue by jbellino-usgs** [geopandas/geopandas#943]
Hello, I have a [shapefile](https://github.com/geopandas/geopandas/files/2961962/box.zip) that uses the ESRI [North America Albers Equal Area Conic](https://epsg.io/102008) projection. When it is read into a GeoDataFrame using the `read_file()` function, the resulting crs is `{'init': 'epsg:102008'}`. When I try to reproject this dataframe, I get `CRSError: Invalid projection: +init=epsg:102008 +type=crs`. When looking through the files in the `pyproj` data directory (e.g. `..\Lib\site-packages\pyproj\data`) I see that the projection "102008" is actually contained within the "esri" file and not the "epsg" file. If I redefine the `crs` to be `{'init': 'esri:102008'}` or using the proj4 string at the link above I am able to reproject.

Simple example to reproduce the error:
```python
import geopandas as gpd
f_in = 'box.shp'
gdf = gpd.read_file(f_in)
projout = '+proj=lcc +lat_1=25 +lat_2=60 +lat_0=42.5 +lon_0=-100 +x_0=0 +y_0=0 +ellps=WGS84 +units=m +no_defs=True'
gdf.to_crs(projout)
```

**Comment by jorisvandenbossche** [comment]
@jbellino-usgs Thanks for opening the issue!

This might depend on the version of fiona you are using. With fiona 1.8.4, I see the following behaviour:

```
In [1]: import geopandas

In [2]: df = geopandas.read_file("../Downloads/box/grid_2km_bbox.shp")

In [4]: df.crs
Out[4]: {}

In [5]: import fiona

In [6]: with fiona.open("../Downloads/box/grid_2km_bbox.shp") as src: 
   ...:     print(src.meta) 

{'driver': 'ESRI Shapefile', 
 'schema': {'properties': OrderedDict([('FID', 'int:11')]), 'geometry': 'Polygon'},
  'crs': {},
  'crs_wkt': PROJCS["North_America_Albers_Equal_Area_Conic",GEOGCS["GCS_North_American_1983",DATUM["North_American_Datum_1983",SPHEROID["GRS_1980",6378137.0,298.257222101]],PRIMEM["Greenwich",0.0],UNIT["Degree",0.0174532925199433],AUTHORITY["EPSG","4269"]],PROJECTION["Albers_Conic_Equal_Area"],PARAMETER["False_Easting",0.0],PARAMETER["False_Northing",0.0],PARAMETER["longitude_of_center",-96.0],PARAMETER["Standard_Parallel_1",20.0],PARAMETER["Standard_Parallel_2",60.0],PARAMETER["latitude_of_center",40.0],UNIT["Meter",1.0],AUTHORITY["Esri","102008"]]'}
```

So not exactly the same as what you observe, but also not great of course, because the `crs` is not set at all.

**Comment by jbellino-usgs** [comment]
I was using fiona 1.8.3 and just upgraded to 1.8.4 and get the same response as before with the slightly incorrect `crs` `{'init': 'esri:102008'}`.

**Comment by jbellino-usgs** [comment]
Here is the full output:

```python
Python 3.7.1 (default, Dec 10 2018, 22:54:23) [MSC v.1915 64 bit (AMD64)]
Type 'copyright', 'credits' or 'license' for more information
IPython 7.3.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import geopandas

In [2]: df = geopandas.read_file('grid_2km.shp')

In [3]: df.crs
Out[3]: {'init': 'epsg:102008'}

In [4]: import fiona

In [5]: fiona.__version__
Out[5]: '1.8.4'

In [6]: with fiona.open('grid_2km.shp') as src:
   ...:     print(src.meta)
   ...:
{'driver': 'ESRI Shapefile', 'schema': {'properties': OrderedDict([('node', 'int:9'), ('row', 'int:9'), ('col', 'int:9'), ('X', 'float:24.15'), ('Y', 'float:24.15'), ('area', 'float:24.15')]), 'geometry': 'Polygon'}, 'crs': {'init': 'epsg:102008'}, 'crs_wkt': 'PROJCS["North_America_Albers_Equal_Area_Conic",GEOGCS["GCS_North_American_1983",DATUM["North_American_Datum_1983",SPHEROID["GRS_1980",6378137,298.257222101]],PRIMEM["Greenwich",0],UNIT["Degree",0.017453292519943295],AUTHORITY["EPSG","4269"]],PROJECTION["Albers_Conic_Equal_Area"],PARAMETER["False_Easting",0],PARAMETER["False_Northing",0],PARAMETER["longitude_of_center",-96],PARAMETER["Standard_Parallel_1",20],PARAMETER["Standard_Parallel_2",60],PARAMETER["latitude_of_center",40],UNIT["Meter",1],AUTHORITY["EPSG","102008"]]'}
```

**Comment by jorisvandenbossche** [comment]
@jbellino-usgs you are on windows, is that correct?

**Comment by jbellino-usgs** [comment]
@jorisvandenbossche Yes, I'm on Windows 10 x64

**Comment by snowman2** [comment]
I tinkered with this a bit:
```
>>> from pyproj import CRS
>>> cc = CRS({'init': 'epsg:102008'})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/snowal/scripts/pyproj/pyproj/crs.py", line 257, in __init__
    super(CRS, self).__init__(projstring)
  File "pyproj/_crs.pyx", line 1283, in pyproj._crs._CRS.__init__
    raise CRSError(
pyproj.exceptions.CRSError: Invalid projection: +init=epsg:102008 +type=crs: (Internal Proj Error: proj_create: crs not found)
>>> cc = CRS('epsg:102008')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/snowal/scripts/pyproj/pyproj/crs.py", line 257, in __init__
    super(CRS, self).__init__(projstring)
  File "pyproj/_crs.pyx", line 1283, in pyproj._crs._CRS.__init__
    raise CRSError(
pyproj.exceptions.CRSError: Invalid projection: epsg:102008: (Internal Proj Error: proj_create: crs not found)
>>> cc = CRS('esri:102008')
>>> cc
<Projected CRS: ESRI:102008>
Name: North_America_Albers_Equal_Area_Conic
Axis Info [cartesian]:
- E[east]: Easting (metre)
- N[north]: Northing (metre)
Area of Use:
- name: North America - Canada and USA (CONUS, Alaska mainland)
- bounds: (-172.54, 23.81, -47.74, 86.46)
Coordinate Operation:
- name: North_America_Albers_Equal_Area_Conic
- method: Albers Equal Area
Datum: North American Datum 1983
- Ellipsoid: GRS 1980
- Prime Meridian: Greenwich
```
And so, the code returned by fiona is incorrect as it is the ESRI and not the EPSG code.

However, using the WKT version of the string works fine:
```
>>> cc = CRS('PROJCS["North_America_Albers_Equal_Area_Conic",GEOGCS["GCS_North_American_1983",DATUM["North_American_Datum_1983",SPHEROID["GRS_1980",6378137.0,298.257222101]],PRIMEM["Greenwich",0.0],UNIT["Degree",0.0174532925199433],AUTHORITY["EPSG","4269"]],PROJECTION["Albers_Conic_Equal_Area"],PARAMETER["False_Easting",0.0],PARAMETER["False_Northing",0.0],PARAMETER["longitude_of_center",-96.0],PARAMETER["Standard_Parallel_1",20.0],PARAMETER["Standard_Parallel_2",60.0],PARAMETER["latitude_of_center",40.0],UNIT["Meter",1.0],AUTHORITY["Esri","102008"]]')
>>> cc
<Projected CRS: PROJCS["North_America_Albers_Equal_Area_Conic",GEO ...>
Name: North_America_Albers_Equal_Area_Conic
Axis Info [cartesian]:
- E[east]: Easting (metre)
- N[north]: Northing (metre)
Area of Use:
- undefined
Coordinate Operation:
- name: unnamed
- method: Albers Equal Area
Datum: North American Datum 1983
- Ellipsoid: GRS 1980
- Prime Meridian: Greenwich
```

This should no longer be an issue when `geopandas` starts using the `pyproj.CRS` class with the WKT string output from fiona.

**Comment by jorisvandenbossche** [comment]
On the other hand: http://epsg.io/102008 does list it (but it indeed seems it is not included in the proj.db looking at the source: no projected systems above 100000 are included)

Something to report to fiona?

**Comment by jorisvandenbossche** [comment]
> Something to report to fiona?

Although I suppose this is actually due to the used epsg source. Because on my machine, I got an empty `{}` for `crs`, probably indicating it did not found an epsg code)

**Comment by snowman2** [comment]
It works with the changes in #998 
![image](https://user-images.githubusercontent.com/8699967/58375786-ce664280-7f20-11e9-94ed-0ea29029ecd4.png)

**Comment by martinfleis** [comment]
Fixed by #1101
